samcli/lib/package/artifact_exporter.py

```diff
@@ -532,6 +533,15 @@ def include_transform_export_handler(template_dict, uploader, parent_dir):
 GLOBAL_EXPORT_DICT = {"Fn::Transform": include_transform_export_handler}
 
 
+def normalized_sam_template(template_yaml):
+    """Takes sam yaml template as input and returns normalized template.
+
+    :param template_yaml: template need to be normalized.
+    :return:
+    """
+    return SamFunctionProvider(template_yaml).template_dict
```

Comment:
Looking at SamFunctionProvider code, looks all you need is the BaseProvider so you don't rely on any unintended side-effects introduced on the template_dict variable by SamFunctionProvider.

```
        self.template_dict = SamBaseProvider.get_template(template_dict, parameter_overrides)
```